apps/showcase/src/components/training/training.component.ts

```diff
@@ -0,0 +1,238 @@
+import {AsyncPipe, JsonPipe, NgComponentOutlet} from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  computed,
+  inject,
+  Input,
+  OnInit,
+  signal,
+  WritableSignal
+} from '@angular/core';
+import {FormsModule} from '@angular/forms';
+import {
+  NgbAccordionModule,
+  NgbDropdown,
+  NgbDropdownItem,
+  NgbDropdownMenu,
+  NgbDropdownToggle
+} from '@ng-bootstrap/ng-bootstrap';
+import {DynamicContentModule, DynamicContentService} from '@o3r/dynamic-content';
+import {DirectoryNode, FileNode, FileSystemTree} from '@webcontainer/api';
+import {firstValueFrom} from 'rxjs';
+import {CopyTextPresComponent} from '../utilities/copy-text';
+import {EditorMode, TrainingProject} from './code-editor-view';
+import {TrainingStepPresComponent} from './training-step';
+
+/** Description of training step */
+interface TrainingStepDescription {
+  /** Step title */
+  title: string;
+  /** URL to step instructions (HTML content) */
+  htmlContentUrl: string;
+  /** Step files configuration */
+  filesConfiguration?: {
+    /** Name of directory in which files will be put */
+    name: string;
+    /** Starting file to be displayed in the project */
+    startingFile: string;
+    /** Commands to run in the project */
+    commands: string[];
+    /** URLs of step project */
+    urls: { [key: string]: string };
+    /** URLs of step solution project */
+    solutionUrls?: { [key: string]: string };
+    /** Mode of the Code Editor */
+    mode: EditorMode;
+  };
+}
+
+/** Training step - consists of its description and its dynamic content */
+type TrainingStep = {
+  /** Description of the training step */
+  description: TrainingStepDescription;
+  /** Dynamic content of the training step */
+  dynamicContent: {
+    /** HTML Content - corresponds to the step instructions */
+    htmlContent: WritableSignal<string>;
+    /** Exercise project of the step */
+    project: WritableSignal<TrainingProject | null>;
+    /** Solution of the exercise project of the step */
+    solutionProject: WritableSignal<TrainingProject | null>;
+  };
+};
+
+/** Resources to get file content */
+type Resource = {
+  /** Resource path */
+  path: string;
+  /** Resource content */
+  content: string;
+};
+
+@Component({
+  selector: 'o3r-training',
+  standalone: true,
+  imports: [
+    AsyncPipe,
+    CopyTextPresComponent,
+    DynamicContentModule,
+    FormsModule,
+    JsonPipe,
+    NgbAccordionModule,
+    NgComponentOutlet,
+    TrainingStepPresComponent,
+    NgbDropdown,
+    NgbDropdownToggle,
+    NgbDropdownMenu,
+    NgbDropdownItem
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  templateUrl: './training.component.html',
+  styleUrl: './training.component.scss'
+})
+export class TrainingComponent implements OnInit {
+  /**
+   * Index of the training step to display
+   */
+  public currentStepIndex = 0;
```

Comment:
Why not using signal for these variables used in the template ?

Reply:
Do we really need it?

Reply:
It was just a question for consistency I do not have strong opinion on that :)